sys/newlib/Makefile.include

```diff
@@ -4,6 +4,9 @@ ifneq (,$(filter newlib_nano,$(USEMODULE)))
   # Test if nano.specs is available
   ifeq ($(shell $(LINK) -specs=nano.specs -E - 2>/dev/null >/dev/null </dev/null ; echo $$?),0)
     USE_NEWLIB_NANO = 1
+        ifeq ($(shell echo "int main(){}; void _exit(int n) {while(1);}" | LC_ALL=C $(CC) -xc - -o /dev/null -lc -specs=nano.specs 2>&1 | grep -q "use of wchar_t values across objects may fail" ; echo $$?),0)
```

Comment:
> Let's fix the pedantic warnings too

done, good idea, IMHO code reviews are mostly against the effects of
lazyness. ;)